torch/distributed/distributed_c10d.py

```diff
@@ -1084,8 +1102,16 @@ def _new_process_group_helper(
     _world.pg_map[pg] = (backend, prefix_store)
     _world.pg_names[pg] = group_name
     _world.pg_backend_config[pg] = str(backend_config)
-    return pg
+    # "" is the default tag for user PGs
+    if pg_tag in [None, ""]:
+        pg_tag = f"ptd:{group_name}"
+        _world.tags_to_pg.setdefault("", []).append(pg)
```

Comment:
Won't this get overwritten by line 1112?

Reply:
This allows us to skip the tag/rank lookup process when using the APIs with an explicit PG.
Maybe it's worth making this work for both tagged and untagged PGs.